### PR TITLE
Added modifier spinner in nutrients editing. Fixes API.

### LINKED ISSUE #2605: Make nutrition input numeric
**Summary:** 

Make the input box for nutritional fact numeric

**Steps to reproduce:** 

if you try to add the nutritional fact, the full keyboard appears instead of the numeric

### PR DESCRIPTION
**Pull request by VaiTon**
Will fix #2605 

Also fix the API problem found by @teolemon.

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java [hunk 1]
```diff
@@ -975,14 +975,15 @@ private void checkFields() {
         if (!editionMode) {
             if (addProductOverviewFragment.areRequiredFieldsEmpty()) {
                 viewPager.setCurrentItem(0, true);
-            } else if (isNutritionDataAvailable() && addProductNutritionFactsFragment.containsInvalidValue()) {
+            } else if (isNutritionDataAvailable() && addProductNutritionFactsFragment.isFieldsInvalid()) {
```
Comment (philippeauriach) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java:978]: weird naming: `is` but `fields` (plural or singular ?) . If the method checks if the fragment has an invalid value, then "hasInvalidValue" or "containsInvalidValue" is fine no ?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java [hunk 2]
```diff
@@ -975,14 +975,15 @@ private void checkFields() {
...
                 viewPager.setCurrentItem(2, true);
             } else {
                 saveProduct();
             }
         } else {
             // edit mode, therefore do not check whether front image is empty or not however do check the nutrition facts values.
-            if ((isNutritionDataAvailable()) && addProductNutritionFactsFragment.containsInvalidValue()) {
+            if (isNutritionDataAvailable() && addProductNutritionFactsFragment.isFieldsInvalid()) {
```
Comment (philippeauriach) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java:985]: same remark

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java (hunk 1 above)
Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java:978]: Right, just wanted to uniform it to the other method. I'm renaming it.

**Comment by teolemon** [comment]
I added closes #2605 @VaiTon assuming it's a full fix

**Comment by teolemon** [comment]
3 little conflicts to solve @VaiTon

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -517,6 +520,9 @@
     <string name="nutrition_carbohydrate">Carbohydrate</string>
     <string name="nutrition_cholesterol">Cholesterol</string>
     <string name="nutrition_energy">Energy</string>
+    <string name="nutrition_energy_kcal">Energy in calories</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml:523]:
```suggestion
    <string name="nutrition_energy_kcal">Energy in calories (kcal)</string>
```

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -517,6 +520,9 @@
...
+    <string name="nutrition_energy_kj">Energy in joules</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml:524]:
```suggestion
    <string name="nutrition_energy_kj">Energy in joules (kJ)</string>
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/nutrition/NutritionProductFragment.java [hunk 5]
```diff
@@ -419,13 +352,84 @@ public void onClick(@NonNull View view) {
...
+            binding.nutrientLevelsCardView.setVisibility(View.GONE);
+            levelItemList.add(new NutrientLevelItem("", "", "", 0));
+            binding.imageGrade.setVisibility(View.GONE);
+        } else {
+            // prefetch the uri
+            customTabActivityHelper = new CustomTabActivityHelper();
+            customTabActivityHelper.setConnectionCallback(this);
+
+            // TODO: Make it international
+            nutritionScoreUri = Uri.parse("https://fr.openfoodfacts.org/score-nutritionnel-france");
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/nutrition/NutritionProductFragment.java:380]: Do we still use it ? If so: https://world-en.openfoodfacts.org/nutriscore

Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/nutrition/NutritionProductFragment.java:380]: Yup when you click on 
immagine
the blue link.

Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/nutrition/NutritionProductFragment.java:380]: Maybe we have a translated string. Let me check.

Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/nutrition/NutritionProductFragment.java:380]: I was convinced it was already international, and handled somewhere else.

Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/nutrition/NutritionProductFragment.java:380]: Done.

**Comment by teolemon** [comment]
Just tried from trunk:
image

**Pull request merged**